deebotozmo/cli.py

```diff
@@ -0,0 +1,439 @@
+    return os.path.expanduser("~/.config/deebotozmo.conf")
+
+
+def config_file_exists() -> bool:
+    """Check if the config file exists."""
+    return os.path.isfile(config_file())
+
+
+def read_config() -> configparser.SectionProxy:
+    """Read and parse the config file."""
+    parser = configparser.ConfigParser()
+    with open(config_file(), encoding="utf-8") as file:
+        parser.read_file(itertools.chain(["[global]"], file), source=config_file())
+    return parser["global"]
+
+
+def write_config(config: dict) -> None:
+    """Create a new config file."""
+    os.makedirs(os.path.dirname(config_file()), exist_ok=True)
+    with open(config_file(), "w", encoding="utf-8") as file:
+        for key in config:
+            file.write(key + "=" + str(config[key]) + "\n")
+
+
+@click.group(chain=True, help="")
+@click.option("--debug/--no-debug", default=False)
+def cli(debug: bool = False) -> None:
+    """Create a click group for nesting subcommands."""
+    logging.basicConfig(format="%(name)-10s %(levelname)-8s %(message)s")
+    logging.root.setLevel(logging.DEBUG if debug else logging.ERROR)
+
+
+@cli.command(name="createconfig", help="logs in with specified email; run this first")
+@click.option("--email", prompt="Ecovacs app email")
+@click.option("--password", prompt="Ecovacs app password", hide_input=True)
+@click.option("--country-code", prompt="your two-letter country code")
+@click.option("--continent-code", prompt="your two-letter continent code")
+@click.option("--verify-ssl", prompt="Verify SSL for API requests", default=True)
+@coro
+async def create_config(
+    email: str, password: str, country_code: str, continent_code: str, verify_ssl: str
+) -> None:
+    """Click subcommand to create a new config."""
+    if config_file_exists() and not click.confirm("overwrite existing config?"):
+        click.echo("Skipping setconfig.")
+        sys.exit(0)
+    config = {}
+    password_hash = md5(password)
+    device_id = md5(str(time.time()))
+    async with aiohttp.ClientSession() as session:
+        try:
+            EcovacsAPI(
+                session,
+                device_id,
+                email,
+                password_hash,
+                continent=continent_code,
+                country=country_code,
+                verify_ssl=verify_ssl,
+            )
+        except ValueError as error:
+            click.echo(error.args[0])
+            sys.exit(1)
+    config["email"] = email
+    config["password_hash"] = password_hash
+    config["device_id"] = device_id
+    config["country"] = country_code.lower()
+    config["continent"] = continent_code.lower()
+    config["verify_ssl"] = verify_ssl
+    write_config(config)
+    click.echo("Config saved.")
+    sys.exit(0)
+
+
+async def run_with_login(*args: Any, **kwargs: Any) -> None:
+    """Execute a command only."""
+    vacbot = DoLogin()
+    await vacbot.run()
+    await vacbot.bot.execute_command(*args, **kwargs)
```

Comment:
When an exception is raised on this line, the goodbye method is never called. Please use try/finally

Example:
```python
    vacbot = DoLogin()
    try:
        await vacbot.run()
        await vacbot.bot.execute_command(*args, **kwargs)
    finally:
        await vacbot.goodbye()
```

Please add try/catch for all methods